cmd/jaeger/internal/integration/datareceivers/jaegerstorage.go

```diff
@@ -37,19 +45,24 @@ func NewJaegerStorageDataReceiver(traceStorage string, storageConfig *jaegerstor
 func (dr *jaegerStorageDataReceiver) Start(tc consumer.Traces, _ consumer.Metrics, _ consumer.Logs) error {
 	ctx := context.Background()
 
-	extFactory := jaegerstorage.NewFactory()
-	ext, err := extFactory.CreateExtension(ctx, extension.CreateSettings{
+	extSet := extension.CreateSettings{
+		ID:                jaegerstorage.ID,
 		TelemetrySettings: componenttest.NewNopTelemetrySettings(),
```

Comment:
I see you still create telemetry settings in this component. My suggestion is that you get them as constructor argument, instead of only getting a logger, and you create the whole telemetry settings in integration.go